--- a/configure.ac
+++ b/configure.ac
@@ -1401,14 +1401,18 @@ case "$host_os" in
 esac
 
 dnl Visibility annotations...
-saved_CFLAGS="$CFLAGS"
-CFLAGS="$CFLAGS -fvisibility=hidden -Werror"
-AC_MSG_CHECKING(whether compiler supports visibility annotations)
-AC_LINK_IFELSE([AC_LANG_PROGRAM([
-                 int foo( void ) __attribute__((visibility("default")));
-               ])],
-               [CFLAGS="$saved_CFLAGS -fvisibility=hidden -D__LIBARCHIVE_ENABLE_VISIBILITY";
-                AC_MSG_RESULT(yes)],
-               [CFLAGS="$saved_CFLAGS"
-                AC_MSG_RESULT(no)])
+if test "x${GENTOO_OPT_PUBLIC_ABI}" != "x1"; then
+  saved_CFLAGS="$CFLAGS"
+  CFLAGS="$CFLAGS -fvisibility=hidden -Werror"
+  AC_MSG_CHECKING(whether compiler supports visibility annotations)
+  AC_LINK_IFELSE([AC_LANG_PROGRAM([
+                   int foo( void ) __attribute__((visibility("default")));
+                 ])],
+                 [CFLAGS="$saved_CFLAGS -fvisibility=hidden -D__LIBARCHIVE_ENABLE_VISIBILITY";
+                  AC_MSG_RESULT(yes)],
+                 [CFLAGS="$saved_CFLAGS"
+                  AC_MSG_RESULT(no)])
+fi
