--- a/configure.ac	2026-09-19 04:54:13.050372558 +0100
+++ b/configure.ac	2026-09-19 04:54:13.062811745 +0100
@@ -1400,6 +1400,7 @@
 esac
 
 dnl Visibility annotations...
+if test "x${GENTOO_OPT_PUBLIC_ABI}" != "x1"; then
 saved_CFLAGS="$CFLAGS"
 CFLAGS="$CFLAGS -fvisibility=hidden -Werror"
 AC_MSG_CHECKING(whether compiler supports visibility annotations)
@@ -1410,6 +1411,7 @@
                 AC_MSG_RESULT(yes)],
                [CFLAGS="$saved_CFLAGS"
                 AC_MSG_RESULT(no)])
+fi
 
 # Ensure test directories are present if building out-of-tree
 AC_CONFIG_COMMANDS([mkdirs],
