--- a/Source/WTF/wtf/ParallelJobsGeneric.h
+++ b/Source/WTF/wtf/ParallelJobsGeneric.h
@@ -34,8 +34,6 @@
 #include <wtf/Platform.h>
 
-#if ENABLE(THREADING_GENERIC)
-
 #include <wtf/Condition.h>
@@ -84,8 +82,6 @@
 } // namespace WTF
 
-#endif // ENABLE(THREADING_GENERIC)
-
 
 #endif // ParallelJobsGeneric_h
--- a/Source/WTF/wtf/ParallelJobsGeneric.cpp
+++ b/Source/WTF/wtf/ParallelJobsGeneric.cpp
@@ -27,8 +27,6 @@
 #include <wtf/ParallelJobsGeneric.h>
 
-#if ENABLE(THREADING_GENERIC)
-
 #include <wtf/NumberOfCores.h>
@@ -132,8 +130,6 @@
 }
 
-#endif // ENABLE(THREADING_GENERIC)
