--- a/Source/WTF/wtf/ParallelJobsGeneric.h
+++ b/Source/WTF/wtf/ParallelJobsGeneric.h
@@ -34,5 +34,3 @@
 #include <wtf/Platform.h>
 
-#if ENABLE(THREADING_GENERIC)
-
 #include <wtf/Condition.h>
@@ -84,6 +82,4 @@
 } // namespace WTF
 
-#endif // ENABLE(THREADING_GENERIC)
-
 
 #endif // ParallelJobsGeneric_h
--- a/Source/WTF/wtf/ParallelJobsGeneric.cpp
+++ b/Source/WTF/wtf/ParallelJobsGeneric.cpp
@@ -27,5 +27,3 @@
 #include <wtf/ParallelJobsGeneric.h>
 
-#if ENABLE(THREADING_GENERIC)
-
 #include <wtf/NumberOfCores.h>
@@ -132,3 +130,1 @@
 }
 
-#endif // ENABLE(THREADING_GENERIC)
