--- a/Source/WTF/wtf/ParallelJobsGeneric.h
+++ b/Source/WTF/wtf/ParallelJobsGeneric.h
@@ -43,9 +43,9 @@
-    WTF_EXPORT_PRIVATE ParallelEnvironment(ThreadFunction, size_t sizeOfParameter, int requestedJobNumber);
+    WTF_EXPORT_DECLARATION ParallelEnvironment(ThreadFunction, size_t sizeOfParameter, int requestedJobNumber);
 
     int numberOfJobs()
     {
         return m_numberOfJobs;
     }
 
-    WTF_EXPORT_PRIVATE void execute(std::span<uint8_t> parameters);
+    WTF_EXPORT_DECLARATION void execute(std::span<uint8_t> parameters);
