--- a/src/include/OpenImageIO/filesystem.h	2026-10-01 03:03:53.290605937 +0100
+++ b/src/include/OpenImageIO/filesystem.h	2026-10-01 03:03:53.305273667 +0100
@@ -455,7 +455,7 @@
     virtual size_t pwrite (const void *buf, size_t size, int64_t offset);
 
     // Return the total size of the proxy data, in bytes.
-    virtual size_t size () const { return 0; }
+    virtual size_t size () const;
     virtual void flush() { }
 
     Mode mode () const { return m_mode; }
--- a/src/libutil/filesystem.cpp	2026-10-01 03:06:41.072491752 +0100
+++ b/src/libutil/filesystem.cpp	2026-10-01 03:06:41.083491613 +0100
@@ -1324,6 +1324,12 @@
     return 0;
 }
 
+OIIO_UTIL_API size_t
+Filesystem::IOProxy::size() const
+{
+    return 0;
+}
+
 size_t
 Filesystem::IOFile::size() const
 {
